File: frontend/src/services/admin/bannerManagementService.js
import { instance as axios } from '../../api/axios';

/**
 * Admin Banner Management Service
 * Handles banner/advertisement management
 */

// Get all banners with filtering and pagination
export const getAllBanners = async (filters) => {
    const response = await axios.post('/api/admin/banners/list', {
        data: {
            keySearch: filters.keySearch && filters.keySearch.trim() !== '' ? filters.keySearch : null,
            status: filters.status && filters.status !== '' ? filters.status : null,
            isPublished: filters.isPublished !== undefined && filters.isPublished !== null ? filters.isPublished : null,
            dateFrom: filters.dateFrom || null,
            dateTo: filters.dateTo || null,
            page: filters.page || 1,
            size: filters.size || 10
        }
    });
    return response;
};

// Get banner details by ID
export const getBannerById = async (id) => {
    const response = await axios.get(`/api/admin/banners/${id}`);
    return response;
};

// Create new banner
export const createBanner = async (bannerData) => {
    const response = await axios.post('/api/admin/banners', {
        data: {
            title: bannerData.title,
            imageUrl: bannerData.imageUrl,
            linkUrl: bannerData.linkUrl,
            position: bannerData.position,
            startDate: bannerData.startDate,
            endDate: bannerData.endDate,
            isPublished: bannerData.isPublished
        }
    });
    return response;
};

// Update banner
export const updateBanner = async (id, bannerData) => {
    const response = await axios.put(`/api/admin/banners/${id}`, {
        data: bannerData
    });
    return response;
};

// Delete banner
export const deleteBanner = async (id) => {
    const response = await axios.delete(`/api/admin/banners/${id}`);
    return response;
};

// Publish banner
export const publishBanner = async (id) => {
    const response = await axios.put(`/api/admin/banners/${id}/publish`);
    return response;
};

// Unpublish banner
export const unpublishBanner = async (id) => {
    const response = await axios.put(`/api/admin/banners/${id}/unpublish`);
    return response;
};

// Update banner status
export const updateBannerStatus = async (id, status) => {
    const response = await axios.put(`/api/admin/banners/${id}/status`, null, {
        params: { status }
    });
    return response;
};

// Bulk delete banners
export const bulkDeleteBanners = async (ids) => {
    const response = await axios.delete('/api/admin/banners/bulk-delete', {
        data: ids
    });
    return response;
};

// Get banner statistics
export const getBannerStatistics = async () => {
    const response = await axios.get('/api/admin/banners/statistics');
    return response;
};

export default {
    getAllBanners,
    getBannerById,
    createBanner,
    updateBanner,
    deleteBanner,
    publishBanner,
    unpublishBanner,
    updateBannerStatus,
    bulkDeleteBanners,
    getBannerStatistics
};

File: backend/src/main/java/vn/fpt/se18/MentorLinking_BackEnd/service/admin/BannerManagementService.java
package vn.fpt.se18.MentorLinking_BackEnd.service.admin;

import vn.fpt.se18.MentorLinking_BackEnd.dto.request.BaseRequest;
import vn.fpt.se18.MentorLinking_BackEnd.dto.request.admin.banner.BannerCreateRequest;
import vn.fpt.se18.MentorLinking_BackEnd.dto.request.admin.banner.BannerFilterRequest;
import vn.fpt.se18.MentorLinking_BackEnd.dto.request.admin.banner.BannerUpdateRequest;
import vn.fpt.se18.MentorLinking_BackEnd.dto.response.BaseResponse;
import vn.fpt.se18.MentorLinking_BackEnd.dto.response.PageResponse;
import vn.fpt.se18.MentorLinking_BackEnd.dto.response.admin.BannerDetailResponse;
import vn.fpt.se18.MentorLinking_BackEnd.dto.response.admin.BannerStatisticsResponse;

import java.util.List;

public interface BannerManagementService {
    BaseResponse<PageResponse<BannerDetailResponse>> getAllBanners(BaseRequest<BannerFilterRequest> request);
    
    BaseResponse<BannerDetailResponse> getBannerById(Long id);
    
    BaseResponse<BannerDetailResponse> createBanner(BaseRequest<BannerCreateRequest> request);
    
    BaseResponse<BannerDetailResponse> updateBanner(Long id, BaseRequest<BannerUpdateRequest> request);
    
    BaseResponse<Void> deleteBanner(Long id);
    
    BaseResponse<Void> publishBanner(Long id);
    
    BaseResponse<Void> unpublishBanner(Long id);
    
    BaseResponse<Void> updateBannerStatus(Long id, String status);
    
    BaseResponse<Void> bulkDeleteBanners(List<Long> ids);
    
    BaseResponse<BannerStatisticsResponse> getStatistics();
}
